modules/resolver-rule-associations/main.tf

```
@@ -0,0 +1,7 @@
+resource "aws_route53_resolver_rule_association" "this" {
+  for_each = var.create ? var.resolver_rule_associations : tomap({})
+
+  name             = try(each.value.name, each.key, null)
```

`each.key` will be always present, but `name` is an optional argument.

```suggestion
  name             = try(each.value.name, null)
```